### ISSUE TITLE
Mumble can't be unminimized after it automatically minimizes

### ISSUE THREAD

**Issue by Bios-Marcel** [mumble-voip/mumble#4310]
**Describe the bug**
When switching workspaces/desktops (Call em what you want) in my desktop environment (KDE), mumble minimizes. Upon unminimizing it again, it goes into an endless loop of minimizing and unminimizing.

**Steps to Reproduce**
1. Use KDE Plasma
2. Start mumble
3. Switch to a different desktop
4. Try unminimizing mumble

**Expected behavior**
It should just show up and not keep minimizing and unminimizing until you kill it.

Alternatively it'd be cool to just have a workaround so that it stops minimizing at all.

**Desktop (please complete the following information):**
 - OD: Ubuntu 20.04 KDE Plasma
 - Version: Mumble 1.3 via flatpak (Same occurs via the 1.3.0 snap and 1.3.0 Ubuntu repo)

**Comment by Krzmbrzl** [comment]
Do you have "hide to tray when minimizing" enabled"? If so, then this would be a duplicate of #3977

**Comment by Bios-Marcel** [comment]
The problem here is, that it minimizes at all when changing the active desktop. Somehow mumble sees "switch desktop" as "minimize" and then hides in tray.

**Comment by Krzmbrzl** [comment]
Hm.. I just tested on KDE Neon and I don't have this behavior. I'm using the current master branch, but afaik there haven't been any changes to the minimize behavior of Mumble recently...

What Qt version are the affected Mumble versions using? I'm using 5.14.2 and my KDE Plasma is 5.19.1

**Comment by Bios-Marcel** [comment]
KDE 5.18.5 with Qt 5.12.8.
But Mumble is a flatpak and uses Qt 5.14.2. Is there any way I can gather additional debug info on this or anything?

**Comment by Bios-Marcel** [comment]
Just now, when i ticked the checkbox and saved, this showed up in the developer console window.
The same errors also appears during the unminimize and maximize flickering.

```
<W>2020-06-21 17:31:49.452 internal error:  void QXcbWindow::setNetWmStateOnUnmappedWindow() called on mapped window
```

**Comment by Krzmbrzl** [comment]
Yeah as I said: The flickering part is "dealt with" in #3977 - iirc this error message was also reported there. It's some sort of Qt internal error. No ide what exactly happens though :shrug: 

> But Mumble is a flatpak and uses Qt 5.14.2. Is there any way I can gather additional debug info on this or anything?

Hm... Idk

----

Okay I just tested again and now I can reproduce the minimize on Desktop-switch. I had to re-enable the "minimize to tray" option in order for this to occur.

Btw: Did you have Mumble maximized before the flickering begins by restoring from tray?

**Comment by Bios-Marcel** [comment]
This happens no matter whether mumble was maximized or not. I'd like to add that I also don't have any custom KWin-rules / scripts defined.

**Comment by Krzmbrzl** [comment]
Interesting. I (and the author of #3977) can only reproduce the flickering if Mumble has been maximized before...

Anyways, I think that the minimize on Desktop-switch is also a relic of the broken minimize to tray implementation that's tracked in #3977, so I tend to still close this issue as a duplicate if that's okay for you :)

Generally my conclusion from looking at the respective code is that this feature needs a rewrite in order to fix the issues associated with it.
As a temporary workaround I'd suggest to disable the minimize to tray feature :point_up:

**Comment by Bios-Marcel** [comment]
Yeah, I am doing that now. Anyway. I'll just reopen this as soon as 1.3.2 comes out if it still doesn't work by then. Thanks for your time.

**Comment by alexandrpaliy** [comment]
Hi.
Thanks for the details here, you've helped me to find the solution (uncheck minimize-to-tray option).

By the way, it's version 1.3.2 already, and the problem still exists - maybe the issue should be reopened.

**Comment by Krzmbrzl** [comment]
The ticket was closed since it is a duplicate of #3977, which is still unresolved :point_up:
